core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -15,179 +15,232 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import io.opencensus.common.Clock;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.common.Timestamp;
+import io.opencensus.stats.Aggregation.Count;
+import io.opencensus.stats.Aggregation.Histogram;
+import io.opencensus.stats.Aggregation.Mean;
+import io.opencensus.stats.Aggregation.Range;
+import io.opencensus.stats.Aggregation.StdDev;
+import io.opencensus.stats.Aggregation.Sum;
+import io.opencensus.stats.AggregationData.CountData;
+import io.opencensus.stats.AggregationData.HistogramData;
+import io.opencensus.stats.AggregationData.MeanData;
+import io.opencensus.stats.AggregationData.RangeData;
+import io.opencensus.stats.AggregationData.StdDevData;
+import io.opencensus.stats.AggregationData.SumData;
+import io.opencensus.stats.MutableAggregation.MutableCount;
+import io.opencensus.stats.MutableAggregation.MutableHistogram;
+import io.opencensus.stats.MutableAggregation.MutableMean;
+import io.opencensus.stats.MutableAggregation.MutableRange;
+import io.opencensus.stats.MutableAggregation.MutableStdDev;
+import io.opencensus.stats.MutableAggregation.MutableSum;
+import io.opencensus.stats.View.Window.Interval;
+import io.opencensus.stats.ViewData.WindowData.CumulativeData;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
 
 /**
  * A mutable version of {@link ViewData}, used for recording stats and start/end time.
  */
-// TODO(songya): update to use refactored implementation.
-abstract class MutableViewData {
+final class MutableViewData {
 
   // TODO(songya): might want to update the default tag value later.
   @VisibleForTesting
   static final TagValue UNKNOWN_TAG_VALUE = TagValue.create("unknown/not set");
 
+  private final View view;
+  private final Map<List<TagValue>, List<MutableAggregation>> tagValueAggregationMap =
+      new HashMap<List<TagValue>, List<MutableAggregation>>();
+  @Nullable private final Timestamp start;
+
+  private MutableViewData(View view, @Nullable Timestamp start) {
```

Comment:
I thought start time won't be necessary to IntervalView, since IntervalView only needs end time and duration. Do you think we still need to record start time for IntervalView?